### ISSUE TITLE
Catching errors on install

### ISSUE THREAD

**Issue by goodroot** [microservice-framework/mfw-cli#7]
`mfw install [service]` takes the following parameters:

  - npm package name, @example/example-package
  - Repository, git@github.com:owner/repo
  - Local path to a directory containing a package, `.`

Here is the current behaviour for an npm package in an empty working directory:

```
~/C/N/play ❯❯❯ mfw install @microservice-framework/microservice-router --save                                                                                    
{ Error: ENOENT: no such file or directory, open '/Users/goodroot/Code/NodeJS/play/package.json'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.readFileSync (fs.js:509:33)
    at MFWCliClass.getPackageJSON (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:1111:33)
    at MFWCliClass.prepareModule (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:424:26)
    at MFWCliClass.install (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:78:8)
    at Command.module.exports.installService (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:1269:10)
    at Command.listener (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/commander/index.js:315:8)
    at emitTwo (events.js:106:13)
    at Command.emit (events.js:191:7)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/goodroot/Code/NodeJS/play/package.json' }
        -       downloading microservice-router
        -       copiyng microservice-router to /Users/goodroot/Code/NodeJS/play/services/microservice-router
        -       installing dependencies for microservice-router
{ Error: ENOENT: no such file or directory, open '/Users/goodroot/Code/NodeJS/play/package.json'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.readFileSync (fs.js:509:33)
    at MFWCliClass.getPackageJSON (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:1111:33)
    at MFWCliClass.setModuleDefaults (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:879:29)
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:803:19
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqWrap.oncomplete (fs.js:123:15)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/goodroot/Code/NodeJS/play/package.json' }

Mongo URL:  (mongodb://localhost:27017/)
Mongo Table:  (routes)
Mongo prefix(db):  (mfw)
Mongo Options (example: ?replicaSet=rs1&slaveOk=true):
IP or hostname of the server:
Admin port:  (3000)
Proxy port:  (3100)
Number of workers:  (2)
How often to clean expired services.:  (30000)
Proxy URL:  (http://localhost:3100/)
Path to API exloper HTML templates.:  (./html)
Do not change:  (service.json)
SECURE_KEY:  (472883efe8037c937ceb66a8c65d8745807bbd8dbe2e2e0b)
PID file path:  (../../pids/microservice-router.pid)
Log file path:  (../../logs/microservice-router.log)

readline.js:982
            throw err;
            ^

Error: ENOENT: no such file or directory, open '/Users/goodroot/Code/NodeJS/play/configs/microservice-router.env'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.writeFileSync (fs.js:1347:33)
    at MFWCliClass.writeEnvFile (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:949:6)
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:821:14
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:336:32
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/async/lib/async.js:154:25
    at assembler (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:333:9)
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:342:32
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:625:5
```

The same, but for a git url:

```
~/C/N/play ❯❯❯ mfw install git@github.com:microservice-framework/microservice-router.git    
                                             
{ Error: ENOENT: no such file or directory, open '/Users/goodroot/Code/NodeJS/play/package.json'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.readFileSync (fs.js:509:33)
    at MFWCliClass.getPackageJSON (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:1111:33)
    at MFWCliClass.prepareModule (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:424:26)
    at MFWCliClass.install (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:78:8)
    at Command.module.exports.installService (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:1269:10)
    at Command.listener (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/commander/index.js:315:8)
    at emitTwo (events.js:106:13)
    at Command.emit (events.js:191:7)
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/goodroot/Code/NodeJS/play/package.json' }
        -       downloading microservice-router.git
        -       copiyng microservice-router.git to /Users/goodroot/Code/NodeJS/play/services/microservice-router.git
        [err]   ENOENT: no such file or directory, open '/Users/goodroot/Code/NodeJS/play/package.json'
        [err]   ENOENT: no such file or directory, lstat '/var/folders/09/m6fcr2gj4vzbklf8lh8ljx600000gn/T/tmp-31294YZ5zcLULz7L/package/'
```

The `package.json` issue goes away when the directory already exists, but the `config/{service_name}.env` file isn't being created and errors out:

```
~/C/N/p/microservice-router ❯❯❯ mfw install .                                              
        -       downloading microservice-router
...
Log file path:  (../../logs/microservice-router.log)
readline.js:982
            throw err;
            ^

Error: ENOENT: no such file or directory, open '/Users/goodroot/Code/NodeJS/play/microservice-router/configs/microservice-router.env'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.writeFileSync (fs.js:1347:33)
    at MFWCliClass.writeEnvFile (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:949:6)
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:821:14
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:336:32
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/async/lib/async.js:154:25
    at assembler (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:333:9)
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:342:32
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:625:5
```

An empty `.env` file is being created within the root directory, but the sym linking isn't taking.

**Comment by Gormartsen** [comment]
@goodroot so it happens in directory where `mfw setup` was not executed or package.json is missing, right ?

**Comment by goodroot** [comment]
@Gormartsen That's right for two of the three cases. 

For this case: 

```
~/C/N/p/microservice-router ❯❯❯ mfw install .                                              
        -       downloading microservice-router
...
Log file path:  (../../logs/microservice-router.log)
readline.js:982
            throw err;
            ^

Error: ENOENT: no such file or directory, open '/Users/goodroot/Code/NodeJS/play/microservice-router/configs/microservice-router.env'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.writeFileSync (fs.js:1347:33)
    at MFWCliClass.writeEnvFile (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:949:6)
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/includes/MFWCliClass.js:821:14
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:336:32
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/async/lib/async.js:154:25
    at assembler (/usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:333:9)
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:342:32
    at /usr/local/lib/node_modules/@microservice-framework/mfw-cli/node_modules/prompt/lib/prompt.js:625:5
```

That is when `mfw setup` has not been executed and the `package.json` file _is_ present.

**Comment by Gormartsen** [comment]
I see. So we need install validation function.
That will check for:
 - is current directory is initiated by mfw setup 
   - package.json exists
   - .env file exists and is empty or contain one line string. If not empty - is file **envContent**.package.json exists
   - directories (configs, services,pids,logs) are in place
- is source package valid mfw package 

If all validations success - process with installation.

Sounds good ?

**Comment by goodroot** [comment]
@Gormartsen That should cover it! 👍

**Comment by Gormartsen** [comment]
@goodroot master version is fixed.
Please test and confirm.

**Comment by goodroot** [comment]
@Gormartsen LGTM. 👍

**Comment by Gormartsen** [comment]
Tnx!
This fix would be released on milestone 1.3.0

**Comment by Gormartsen** [comment]
Looks like this one is done.
I will cover this cases in tests
